test/test_magnet.cpp

```diff
@@ -224,6 +224,23 @@ TORRENT_TEST(magnet)
 	p2 = s->abort();
 }
 
+TORRENT_TEST(parse_escaped_hash_parameter)
+{
+	error_code ec;
+	add_torrent_params p;
+	parse_magnet_uri("magnet:?xt=urn%3Abtih%3Acdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcd", p, ec);
```

Comment:
I agree (that supporting escaped characters should be supported). What I'm hesitant about is to have everyone pay for this support, that's only needed very rarely (presumably). Can you think of a way to only copy the string in case there is any unescaping to be done?
The first thing that comes to my mind would be something like this, but maybe there's simpler/better way:

```
std::string info_hash_copy;
string_view btih = url_has_argument(uri, "xt");
// ... existing code
if (btih.find('%') != string_view::npos)
{
   info_hash_copy = unescape_string(btih);
   btih = info_hash_copy;
}
// ... more existing code
```